components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/IdentityKeyStoreResolver.java

```diff
@@ -0,0 +1,589 @@
+                            .getDescription(), keyStoreName), e);
+        }
+    }
+
+    private void parseIdentityKeyStoreMappingConfigs() {
+
+        OMElement keyStoreMappingsElem = IdentityConfigParser.getInstance().getConfigElement(
+                IdentityKeyStoreResolverConstants.CONFIG_ELEM_SECURITY).getFirstChildWithName(
+                IdentityKeyStoreResolverUtil.getQNameWithIdentityNameSpace(
+                        IdentityKeyStoreResolverConstants.CONFIG_ELEM_KEYSTORE_MAPPING));
+
+        // Parse OAuth KeyStore Mapping.
+        OMElement oauthKeyStoreMapping = keyStoreMappingsElem.getFirstChildWithName(
+                IdentityKeyStoreResolverUtil.getQNameWithIdentityNameSpace(
+                        IdentityKeyStoreResolverConstants.CONFIG_ELEM_OAUTH));
+        if (oauthKeyStoreMapping != null) {
+            addKeyStoreMapping(InboundProtocol.OAUTH, oauthKeyStoreMapping);
+        }
+
+        // Parse WS-Trust KeyStore Mapping.
+        OMElement wsTrustKeyStoreMapping = keyStoreMappingsElem.getFirstChildWithName(
+                IdentityKeyStoreResolverUtil.getQNameWithIdentityNameSpace(
+                        IdentityKeyStoreResolverConstants.CONFIG_ELEM_WS_TRUST));
+        if (wsTrustKeyStoreMapping != null) {
+            addKeyStoreMapping(InboundProtocol.WS_TRUST, wsTrustKeyStoreMapping);
+        }
+
+        // Parse WS-Federation KeyStore Mapping.
+        OMElement wsFedKeyStoreMapping = keyStoreMappingsElem.getFirstChildWithName(
+                IdentityKeyStoreResolverUtil.getQNameWithIdentityNameSpace(
+                        IdentityKeyStoreResolverConstants.CONFIG_ELEM_WS_FEDERATION));
+        if (wsFedKeyStoreMapping != null) {
+            addKeyStoreMapping(InboundProtocol.WS_FEDERATION, wsFedKeyStoreMapping);
+        }
+    }
+
+    private void addKeyStoreMapping(InboundProtocol protocol, OMElement keyStoreMapping) {
+
+        // Parse keystore name
+        OMElement keyStoreNameElement = keyStoreMapping.getFirstChildWithName(
+                IdentityKeyStoreResolverUtil.getQNameWithIdentityNameSpace(
+                        IdentityKeyStoreResolverConstants.ATTR_NAME_KEYSTORE_NAME));
+        if (keyStoreNameElement == null || keyStoreNameElement.getText().isEmpty()) {
+            LOG.error("Error occurred when reading KeyStoreMapping configuration. KeyStoreName value null.");
+            return;
+        }
+        String keyStoreName = keyStoreNameElement.getText();
+
+        // Parse UseInAllTenants config
+        OMElement useInAllTenantsElement = keyStoreMapping.getFirstChildWithName(
+                IdentityKeyStoreResolverUtil.getQNameWithIdentityNameSpace(
+                        IdentityKeyStoreResolverConstants.ATTR_NAME_USE_IN_ALL_TENANTS));
+        if (useInAllTenantsElement == null || useInAllTenantsElement.getText().isEmpty()) {
+            LOG.debug("use_in_all_tenants config null for " + protocol.toString() + ". Using default value as false.");
```

Comment:
Add the debug enabled check

Reply:
Fixed in https://github.com/wso2/carbon-identity-framework/pull/5794/commits/979357d8ffcf447df8ffc6585c80efa4d596022a